File: en-game/front-end-admin/src/app/main/dto/Product/ProductExportExcelDTO.ts
export class ProductExportExcelDTO {
    index: number;
    name: string;
    categoryName: string;
    newPrice: number;
    oldPrice: number;
    discount: number;
    cartCount: number;
    weight: number;
    size: number;
    availibilityCount: number;
    createdDate: string;
    updatedDate: string;
    active: boolean;

    constructor(index: number, name: string, categoryName: string, newPrice: number,
        oldPrice: number, discount: number, cartCount: number, weight: number, size: number,
        availibilityCount: number, createdDate: string, updatedDate: string, active: boolean) {
        this.index = index;
        this.name = name;
        this.categoryName = categoryName;
        this.newPrice = newPrice;
        this.oldPrice = oldPrice;
        this.discount = discount;
        this.cartCount = cartCount;
        this.weight = weight;
        this.size = size;
        this.availibilityCount = availibilityCount;
        this.createdDate = createdDate;
        this.updatedDate = updatedDate;
        this.active = active;
    }
}

File: en-game/back-end/admin-service/src/main/java/com/endgame/adminservice/dto/product/ProductExportExcelDTO.java
package com.endgame.adminservice.dto.product;

import com.endgame.adminservice.dto.category.CategoryListProductDTO;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class ProductExportExcelDTO {
    private Long id;
    private String name;
    private CategoryListProductDTO category;
    private BigDecimal newPrice = BigDecimal.ZERO;
    private BigDecimal oldPrice = BigDecimal.ZERO;
    private BigDecimal discount = BigDecimal.ZERO;
    private Long cartCount = 0L;
    private BigDecimal weight = BigDecimal.ZERO;
    private Long availibilityCount = 0L;
    private String createdDate;
    private String size;
    private String updatedDate = createdDate;
    private boolean active;
}
